R/topTaxa.R

```diff
@@ -0,0 +1,90 @@
+#' Returns top N \dQuote{FeatureID}
+#'
+#' Basic extractor function for extracting top N \dQuote{FeatureID}
+#' from assay for a user specified \dQuote{SampleID} in a
+#' \code{\link[=SummarizedExperiment-class]{SummarizedExperiment}}
+#' object.
+#'
+#' @param x A
+#'  \code{\link[=SummarizedExperiment-class]{SummarizedExperiment}} object.
+#'
+#' @param top Numeric value, how many top taxa to return. Default return top five taxa.
+#'
+#' @param method Specify the method to determine top taxa. Either sum, mean or median.
+#'               Default is 'mean'
+#'
+#' @param abund_values a \code{character} value to select an
+#'   \code{\link[SummarizedExperiment:SummarizedExperiment-class]{assayNames}}
+#'
+#' @importFrom DelayedMatrixStats rowSums2 rowMeans2 rowMedians
+#'
+#' @return A vector of the top N \dQuote{FeatureID}
+#' @name topTaxa
+#'
+#' @author
+#' Sudarshan A. Shetty
+#'
+#' @examples
+#' data(GlobalPatterns, package = "MicrobiomeExperiment")
+#'
+#' top_taxa <- topTaxa(GlobalPatterns,
+#'                     method="mean",
+#'                     top=5,
+#'                     abund_values="counts")
+#'
+NULL
+
+#' @rdname topTaxa
+#'
+#' @export
+
+setGeneric("topTaxa", signature = "x",
+           function(x, top, method, abund_values)
+               standardGeneric("topTaxa"))
+
+.check_max_taxa <- function(x, top, abund_values){
```

Comment:
a check that top is actually an integer value is missing. If `top = "whatever"` the function does not work.